plugins/plugin-chart-table/src/index.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { t } from '@superset-ui/translation';
+import { ChartMetadata, ChartPlugin } from '@superset-ui/chart';
+import transformProps from './transformProps';
+import thumbnail from './images/thumbnail.png';
+import controlPanel from './controlPanel';
+
+export * from './types';
+
+const metadata = new ChartMetadata({
+  canBeAnnotationTypes: ['EVENT', 'INTERVAL'],
+  description: '',
+  name: t('Table'),
+  thumbnail,
+  useLegacyApi: true,
```

Comment:
Do you think it would be possible to use the new API for this? If there are any features that are missing I'm sure we could add them. Migrating the table viz to the new endpoint would be a really big step forward, as it's one of the more complex plugins in `viz.py`

Reply:
I took a dab earlier but it feels more complicated than I thought. I'd prefer to merge the frontend change first, then replace the API. The API migration seems to worth its own work as there might be need to change the query client as well.